src/dotty/tools/dotc/core/TypeErasure.scala

```diff
@@ -341,11 +338,16 @@ class TypeErasure(isJava: Boolean, isSemi: Boolean, isConstructor: Boolean, wild
 
   private def eraseArray(tp: RefinedType)(implicit ctx: Context) = {
     val defn.ArrayType(elemtp) = tp
+    val arrayErasure = erasureFn(isJava, isSemi = false, isConstructor, wildcardOK)(_)
```

Comment:
```
def arrayErasure(tp: Type) = erasureFn(isJava, isSemi = false, isConstructor, wildcardOK)(tp)
``` 
will be compiled to a method on this class.
```
erasureFn(isJava, isSemi = false, isConstructor, wildcardOK)(_)
```
is an anonymous Function1 subclass, which will be compiled into separate lifted class.


And in general, I believe that code were functions are functions, instead of variables, is easier to follow.

Also, as you use `arrayErasure` only in one of 4 cases, making it a `val` allocates a class even if you are not going to use it.

To sum up, here `def` is:
 - smaller in tems of bytecode footprint(method vs class in bytecode);
 - faster(no need to allocate a class if arrayErasure is not used);
 - easier to follow.